models/src/main/protobuf/CasperMessage.proto

```diff
@@ -178,6 +178,11 @@ message BodyProto {
   repeated ProcessedDeployProto deploys      = 2;
   repeated ProcessedSystemDeployProto systemDeploys = 3;
   bytes                    extraBytes   = 4;
+  repeated RejectedDeployProto rejectedDeploys = 5;
```

Comment:
Do we need additional type or it can be just `repeated bytes`. Do you know if there is a difference in serialized data?

Reply:
Original plan is to also put the blockHash in the RejectedDeployProto. But it is difficult to get blockHash with current structure.

Reply:
What do you think adding blockHash?

I got this idea because one deploy can possible be in two blocks(or more) when the deploy is not finalized. This design is too sticky.

Reply:
For now let's keep it without block hash, but I think we will need additional index for quick lookup of deploy status.

Reply:
I guess we did not talk about how rejected deploy will actually be used, so just want to put some thoughts.

This rejected deploy will be involved when: 
1) when preparing unfinalized deploys to be put in new block. We might cancel deploy that is already found in scope, because of rejection 
and 
2) when removing deploys from deploy store on finalize. We might not remove deploy if rejection is found in the scope of DAG we are finalizing on.